### DISCUSSION TITLE
Suggestion: (Server) Close window shouldn't shut down the server

### DISCUSSION THREAD

**Discussion by juanriccio** [jamulussoftware/jamulus#3142] [Ideas]
Hello, I am running a jamulus server nearly 24/7, but when I open its window and then close it, the server shuts down. It is a mistake I often do, sometimes in the middle of a session.

I suggest that closing the window should let the process live, just closing the window, This is customary, for example, in Windows, where user daemon processes stay in the system tray until explicitly ended. An explicit "Stop server" button could be added to that window. Actually, my server runs on Linux, but I'd love the same UI paradigm - especially for a server that ideally is meant to run for days on end, and some times gets shut off by op error  :-)

Thanks for your attention!

**Comment by rdica** [comment]
Are you aware you don't need a GUI to run the linux server?

**Comment by juanriccio** [comment]
Yes, I was half aware: I read about that when installing on linux for the first time. I opted for the GUI version because it feels simpler to me. The problem is that when I close the window, it doesn't revert to a GUI-less process. This would be OK, as long as I could reopen the GUI with some ccmmand. Instead, it just kills the whole server.

**Reply by rdica** [reply]
Closing the server window is supposed to kill the process. This is same behaviour on all supported OSes. You have a choice of running the server with a GUI, or without. You can minimize the window if you choose to run the server in GUI mode.

**Comment by pljones** [comment]
> Hello, I am running a jamulus server nearly 24/7, but when I open its window and then close it, the server shuts down. It is a mistake I often do, sometimes in the middle of a session.
> 
> I suggest that closing the window should let the process live, just closing the window, This is customary, for example, in Windows, where user daemon processes stay in the system tray until explicitly ended. An explicit "Stop server" button could be added to that window. Actually, my server runs on Linux, but I'd love the same UI paradigm - especially for a server that ideally is meant to run for days on end, and some times gets shut off by op error :-)
> 
> Thanks for your attention!

The problem on Linux is that there are multiple Desktop standards with different approaches, not all allowing an application to put a notification icon in the task bar, as we do for the Windows version.  As @rdica says, if you're running the GUI on your Linux desktop, just iconise the application window - the application will keep running.

If you want the application to run when you're not logged in, then you'll need it to run as a separate user.  The supplied process for running a headless server Linux explains this.
